### PR TITLE
Investment requirements

### PR DESCRIPTION
**Pull request by tyom**
- [x] Investment requirements key/value table
- [x] Investment requirements form
  - [x] Ability to add basic data
  - [x] Conditional field and progressive disclosure for UK company to be invested in

requirements ng

### CODE REVIEW

File: config/nunjucks/filters.js [hunk 1]
```diff
@@ -20,6 +20,8 @@ const filters = {
 
   keys: _.keys,
 
+  map: _.map,
```
Comment (teneightfive) [inline config/nunjucks/filters.js]: I think I've removed underscore as a default dependency and just used destructuring like above in another PR. Maybe rebase on develop to make sure you have the latest changes.

Comment (feedmypixel) [inline config/nunjucks/filters.js]: Maybe not part of this PR but can we call this `lodash`? rather than `_` to align to the rest of the app

File: src/controllers/investment/form.middleware.js [hunk 2]
```diff
@@ -227,11 +243,37 @@ function interactionDetailsFormPostMiddleware (req, res, next) {
...
+      res.locals.form.errors = err.error
+      res.locals.form.state = req.body
+
+      next()
+    } else {
+      next(err)
+    }
+  })
+}
+
```
Comment (feedmypixel) [inline src/controllers/investment/form.middleware.js]: as you mentioned it is starting to feel like these could be split out into separate files or feature groupings. Something like: `/middleware/form/<name>.js` folder/filenames. Maybe easier as a different PR

Comment (tyom) [inline src/controllers/investment/form.middleware.js]: I'd say a different PR would be more appropriate.

File: config/nunjucks/filters.js [hunk 3]
```diff
@@ -19,10 +20,9 @@ const filters = {
   stringify: JSON.stringify,
 
   values,
-
   keys,
-
```
Comment (teneightfive) [inline config/nunjucks/filters.js]: Any reason for removing the breathing space? Shouldn't we be consistent and remove the line between all methods in this file in that case?

Comment (tyom) [inline config/nunjucks/filters.js]: I thought it looked odd with shorthanded properties.

```js
  stringify: JSON.stringify,

  values,
  
  keys,
  
  flatten,
  
  map,

  highlight: (string, searchTerm) => { }
```

Too much breathing space.

File: src/controllers/investment/shared.middleware.js [hunk 4]
```diff
@@ -29,10 +29,11 @@ function getProjectDetails (req, res, next, id = req.params.id) {
   Q.spawn(function * () {
     try {
       const projectData = yield getInvestmentProjectSummary(req.session.token, req.params.id)
-      res.locals.valueData = yield getInvestmentValue(req.session.token, req.params.id)
+      const valueData = yield getInvestmentValue(req.session.token, req.params.id)
       res.locals.requirementsData = yield getInvestmentRequirements(req.session.token, req.params.id)
       res.locals.teamData = yield getInvestmentTeam(req.session.token, req.params.id)
       res.locals.projectData = projectData
+      res.locals.valueData = valueData
```
Comment (tyom) [inline src/controllers/investment/shared.middleware.js]: `valueData` is cached because it's used further down in `projectStatus` locals.
